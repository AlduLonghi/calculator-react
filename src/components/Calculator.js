import { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../style/App.css';
import Navbar from './shared/Navbar';

const Calculator = () => {
  const [state, setState] = useState({ total: '', next: '', operation: '' });
  const { total } = state;
  const { next } = state;
  const { operation } = state;

  const handleClick = btnName => {
    setState(prevState => ({ ...calculate(prevState, btnName) }));
  };

  return (
    <>
      <Navbar />
      <div className="calculator-body">
        <Display data={{ total, next, operation }} />
        <ButtonPanel onClick={handleClick} />
      </div>
    </>
  );
};

export default Calculator;
