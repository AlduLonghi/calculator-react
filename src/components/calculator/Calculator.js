import CalculatorBody from './CalculatorBody';
import '../../style/App.css';
import calculatorImg from '../../assets/calculator.svg';

const Calculator = () => (
  <div className="calculator-component d-flex">
    <div className="calc-text-img d-flex">
      <h3>Let&apos;s do some math!</h3>
      <img src={calculatorImg} alt="calculator" />
    </div>
    <CalculatorBody />
  </div>
);

export default Calculator;
