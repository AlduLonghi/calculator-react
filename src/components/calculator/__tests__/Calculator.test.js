import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('Snapshot test for Calculator component', () => {
    const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})