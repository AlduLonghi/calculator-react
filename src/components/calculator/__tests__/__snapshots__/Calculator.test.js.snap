// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Calculator component 1`] = `
<div
  className="calculator-component d-flex"
>
  <div
    className="calc-text-img d-flex"
  >
    <h3>
      Let's do some math!
    </h3>
    <img
      alt="calculator"
      src="calculator.svg"
    />
  </div>
  <div
    className="calculator-body"
  >
    <div
      className="display d-flex"
    >
      <p>
          
      </p>
    </div>
    <div>
      <div
        className="button-row d-flex"
      >
        <button
          className="button button-spc"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button button-spc"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button button-spc"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button button-spc"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="button-row d-flex"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="button-row d-flex"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="button-row d-flex"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="button-row d-flex"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button button-equal"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
